x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -352,6 +356,23 @@ public Query termQuery(Object value, SearchExecutionContext context) {
             throw new IllegalArgumentException(CONTENT_TYPE + " fields do not support term query");
         }
 
+        @Override
+        public Query existsQuery(SearchExecutionContext context) {
+            if (getEmbeddingsField() == null) {
+                return new MatchNoDocsQuery();
+            }
+
+            // Do the equivalent of a nested query with an exists query for the embeddings field
+            String nestedFieldPath = getChunksFieldName(name());
+            BitSetProducer parentFilter = new QueryBitSetProducer(Queries.newNonNestedFilter(IndexVersion.current()));
+            return new ESToParentBlockJoinQuery(
```

Comment:
I wonder if there's a more direct way of doing this. As this method returns a Lucene query, I used the same that NestedQueryBuilder returns.

Reply:
The other way I can think of would be to add the original field during indexing using `DocumentParserContext#addToFieldNames`. That would allow the default exists method to work at the cost of an additional inverted list in the `FieldNamesFieldMapper` field.

Reply:
That would be really simple to do and would work OOTB. I understand that this is a last resort when we don't have doc_values / norms, so we should probably explore the option to delegate the query to the underlying field mappers, right?

Reply:
Well it's a tradeoff. I am fine with the approach in this PR since it doesn't require extra storage.